hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,236 @@
+
+#include "hoomd/ForceCompute.h"
+#include "NeighborList.h"
+#include "hoomd/Index1D.h"
+#include "hoomd/GlobalArray.h"
+#include "hoomd/ManagedArray.h"
+#include <memory>
+
+/*! \file TablePotential.h
+    \brief Declares the TablePotential class
+*/
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+#ifdef __HIPCC__
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+#ifndef __TABLEPOTENTIAL_H__
+#define __TABLEPOTENTIAL_H__
+
+//! Computes the potential and force on each particle based on values given in a table
+/*! \b Overview
+    Pair potentials and forces are evaluated for all particle pairs in the system within the given cutoff distances.
+    Both the potentials and forces** are provided the tables V(r) and F(r) at discreet \a r values between \a rmin and
+    \a rmax. Evaluations are performed by simple linear interpolation, thus why F(r) must be explicitly specified to
+    avoid large errors resulting from the numerical derivative. Note that F(r) should store - dV/dr.
+
+    \b Table memory layout
+
+    V(r) and F(r) are specified for each unique particle type pair. Three parameters need to be stored for each 
+    potential: rmin, rmax, and dr, the minimum r, maximum r, and spacing between r values in the table respectively. 
+    V(0) is the value of V at r=rmin. V(i) is the value of V at r=rmin + dr * i where i is chosen such that r >= rmin
+    and r <= rmax. V(r) for r < rmin and > rmax is 0. The same goes for F. Thus V and F are defined between the region
+    [rmin,rmax], inclusive.
+
+    \b Interpolation
+    Values are interpolated linearly between two points straddling the given r. For a given r, the first point needed, i
+    can be calculated via i = floorf((r - rmin) / dr). The fraction between ri and ri+1 can be calculated via
+    f = (r - rmin) / dr - Scalar(i). And the linear interpolation can then be performed via V(r) ~= Vi + f * (Vi+1 - Vi)
+    \ingroup computes
+*/
+class EvaluatorPairTable
+    {
+    public:
+        //! Define the parameter type used by this pair potential evaluator
+        struct param_type
+            {
+            unsigned int width;
+            Scalar rmin;
+            ManagedArray<Scalar> V_table;
+            ManagedArray<Scalar> F_table;
+
+            #ifdef ENABLE_HIP
+            //! Set CUDA memory hints
+            void set_memory_hint() const
+                {
+                // default implementation does nothing
+                }
+            #endif
+
+            #ifndef __HIPCC__
+            param_type() : width(0), rmin(0.0), V_table({}), F_table({}) {}
+
+            param_type(pybind11::dict v)
+                {
+                width = v["width"].cast<unsigned int>();
+                rmin = v["r_min"].cast<Scalar>();
+                unsigned int align_size = 8; //for AVX
```

Comment:
We aren't using AVX in the table potentials. You can allocate the arrays with size `width` and remove the `, 32` and all the alignment size computations.